### PR TITLE
release: Remove gitian

### PR DESCRIPTION
**Pull request by fanquake**
Note that this doesn't yet touch any glibc back compat related code.

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK.

**Comment by laanwj** [comment]
Concept ACK, but after the 22.0 release. I think it makes sense to wait until we have done a successful GUIX release before dropping all gitian support.

**Comment by hebasto** [comment]
@laanwj 
> but after the 22.0 release.

Not arguing, just noticing that in the current state gitian builds do not pass symbol check. If gitian remains in the repo for 22.0 release, it must be fixed with #22281, #22287, #22305, #22318, #22321.

Hope to discuss this topic [today](https://www.erisian.com.au/bitcoin-core-dev/log-2021-07-03.html#l-112).

**Comment by prayank23** [comment]
Not sure if there is a typo in PR title. Maybe there is one extra `I` in prefix if used for Work In Progress.

**Comment by MarcoFalke** [comment]
I don't think it matters when this is merged. Even if this is removed for 22.0, calling `git revert 2338298190` if needed should be trivial because there can't be any conflicts.

**Comment by fanquake** [comment]
> If gitian remains in the repo for 22.0 release, it must be fixed with #22281, #22287, #22305, #22318, #22321.

I would very much rather avoid all of those PRs. Even more back compat code, and hacking up our dependencies to fix a system that currently, we aren't even going to use for the release, and if gitian is going to be removed then it'd all just have to be reverted / removed.

**Comment by achow101** [comment]
Why not have gitian do the guix build? I know at least one person has mentioned previously that they preferred that.

**Comment by MarcoFalke** [comment]
> Why not have gitian do the guix build?

Someone would have to write code for that.

**Comment by laanwj** [comment]
> Not arguing, just noticing that in the current state gitian builds do not pass symbol check. If gitian remains in the repo for 22.0 release, it must be fixed with #22281, #22287, #22305, #22318, #22321.

Agree, I am aware it is broken at the moment and I'm not arguing that it should be fixed up right now, I'd just prefer to wait with removing gitian completely until a successful release with guix has been done. 

Once we've been through the whole process and no unexpected snags turn up (including with people testing the rcs), merging this is fine.

File: ci/test/00_setup_env_win64.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@
 export LC_ALL=C.UTF-8
 
 export CONTAINER_NAME=ci_win64
-export DOCKER_NAME_TAG=ubuntu:20.04  # Check that Focal can cross-compile to win64 (Focal is used in the gitian build as well)
+export DOCKER_NAME_TAG=ubuntu:20.04  # Check that Focal can cross-compile to win64
```
Comment (MarcoFalke) [inline ci/test/00_setup_env_win64.sh:10]: Might be good to replace with a comment to explain why Focal was picked:

LTS that offers a compiler closest to the guix-win64 compiler

**Comment by laanwj** [comment]
Code review ACK ab9c34237ab7b056394e0bd1f7cb131ffd95754c

File: ci/test/00_setup_env_win64.sh (hunk 1 above)
Comment (fanquake) [reply ci/test/00_setup_env_win64.sh:10]: I  can address this along with another change

**Pull request merged**